continuum/tasks/task_set.py

```diff
@@ -107,7 +110,7 @@ def get_sample(self, index: int) -> np.ndarray:
         """
         x = self._x[index]
 
-        if self.data_type == "image_path":
+        if self.data_type in ("image_path", "segmentation"):
```

Comment:
pourquoi segmentation et image_path sont deux object de type different ici?

Reply:
While for both `x` are actually image paths, for the `segmentation` type the handling is different in the task set and we rely on the data type for this.

Plus note that `y` is an array of integer when `data_type=image_path`, but an array of paths when `data_type=segmentation`.

Reply:
Ok je vois, je reviens a mon idee de faire une class task_set differente pour gerer les donnee de segmentation ca eviterai d'avoir un cree un data_type comme ca un peu arbitraire et pas evident a comprendre de l 'exterieur.

Reply:
I'm not really sure about that.

- exterior doesn't need per se to look at the internal, but it's quite straightforward I think. Although we could add in-code doc to explain that.
- currently the additional code to handle different type is quite minimal
- I'm afraid we may confuse more users by having different kind of TaskSets.

Reply:
segmentation and classification are two different types of tasks, it makes sens to use different tasksets for both.
A taskset represent a task, different types of taskset for different types of task is quite normal.
I like to have structure in projects and not put to much features or use cases for one class.
I prefer it with two different objects, if you don't have a strong opinion about it and since it is a personal opinion from me I can do it if you want.

Reply:
#104